engine/src/main/java/io/zeebe/engine/processing/job/FailProcessor.java

```diff
@@ -33,9 +38,21 @@ private void acceptCommand(
       final TypedRecord<JobRecord> command, final CommandControl<JobRecord> commandControl) {
     final long key = command.getKey();
     final JobRecord failedJob = state.getJob(key);
-    failedJob.setRetries(command.getValue().getRetries());
+    final long retryBackOff = command.getValue().getRetryBackOff();
+    final int retries = command.getValue().getRetries();
+    failedJob.setRetries(retries);
+    failedJob.setRetryBackOff(retryBackOff);
     failedJob.setErrorMessage(command.getValue().getErrorMessageBuffer());
-    state.fail(key, failedJob);
+    if (retryBackOff != 0) {
+      state.failForBackoff(key, failedJob);
+      actorControl.runDelayed(
+          Duration.ofMillis(retryBackOff),
+          () -> {
+            state.activable(key, failedJob);
+          });
```

Comment:
Nice idea :sweat_smile: 

But I think that using the `actorControl.runDelayed()` doesn't scale well. The other problem is that it is not persisted in the state. On restart or leader change, the job will no be activated again.

We should follow the same way as `DueDateTimerChecker`. I guess that we can abstract and reuse most of its scheduling logic.